uwblueprint/supportive-housing
# Create buildings table

# Helen's PR

## GitHub Issue link

[Create Buildings Table](https://github.com/uwblueprint/supportive-housing/issues/83)



## Implementation description
* Create Buildings table
* Update the Residents and Log_records table to have building_id as a foreign key for the buildings table
* Update the Residents and Log_records APIs to reflect this change
* Add the three buildings on the SHOW website to the Buildings table whenever we initialize the database 



## Steps to test
1. Check that Buildings table has the three building instances
2. Check that Residents API still works (please refer to the notion docs)
3.  Check that Log_records API still works (please refer to the notion docs)



## What should reviewers focus on?
* preexisting API

# Kelly's Edits

- on backend, fixed apis to return `building_id` instead of `building`
- in frontend, fixed API calls to use `building_id` instead of `building`
- fixed adding, editing, filtering, counting log records to have parameter of `building_id`
- fixed adding and editing resident to have parameter of `building_id`
- edit existing scripts and added new one to create buildings


## Steps to test
1. Drop your database and create it again (so you can get the new buildings table)
2. Run the scripts in repo read.me AND this new script `bash ./seeding/create-buildings.sh`
3. Add residents
4. Edit residents
5. Filter log records
6. Add log records
7. Edit log records


## What should reviewers focus on?
* check functionality
* check if script works


## Checklist
